### PR TITLE
Adding HA documentation

### PR DESCRIPTION
**Pull request by denisok**
https://jira.percona.com/browse/PMM-10901

### CODE REVIEW

File: docs/setting-up/server/ha.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
+# High Availability Cluster for PMM Server
+
+High Availability for PMM Server would be achieved in 3 phases:
+
+1. Prevent data loss with VM integration for short outages
+2. HA data sources
+3. Clustered PMM
+
+The first phase improves PMM Client to survive broken connections and cache metrics that are configured for the `push` mode. This phase relies on external service to restore PMM Server, or for manual interventions to bring PMM Server and/or its connectivity back to a healthy state.
```
Comment (rasika-chivate) [inline docs/setting-up/server/ha.md:9]:
```suggestion
The first phase improves PMM Client to survive broken connections and cache metrics configured for the `push` mode. This phase relies on external service to restore PMM Server or for manual interventions to bring PMM Server and/or its connectivity back to a healthy state.
```

File: docs/setting-up/server/ha.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+The third phase would add Cluster support to PMM Server so it could run in cluster mode with other PMM Servers and possibly load balance the load as well.
+
+The first two phases require PMM Server to be managed by some services or manually. In case of failures caused by network outages, resource saturation, hardware failures, operating system crashes, or unexpected reboots, something or someone would restore PMM Server and get it up and running so PMM Clients and users could connect to it. Also, it is required to have additional external monitoring and alerting to detect PMM Server availability and notify the PMM administrator to take some actions.
+
+The first two phases would improve the availability of PMM Server (or it's data) but don't provide any management capabilities to restore PMM Server to a desired healthy state. Currently, the PMM administrator should do it manually and ensure that such resources are available:
```
Comment (rasika-chivate) [inline docs/setting-up/server/ha.md:17]: To sound more positive we can say 

```suggestion
The first two phases would improve the availability of the PMM Server (or its data) but need to provide management capabilities to restore PMM Server to a desired healthy state. Currently, the PMM administrator has to do it manually and ensure that such resources are available:
```

File: docs/setting-up/server/ha.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+- the network which connects PMM Server, PMM Clients, and PMM users
```
Comment (rasika-chivate) [inline docs/setting-up/server/ha.md:19]:
```suggestion
- The network which connects PMM Server, PMM Clients, and PMM users
```

File: docs/setting-up/server/ha.md [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+- storage where PMM Server store the data
```
Comment (rasika-chivate) [inline docs/setting-up/server/ha.md:20]:
```suggestion
- Storage where PMM Server store the data
```

File: docs/setting-up/server/ha.md [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+- PMM Server service is running (docker, podman, VM)
```
Comment (rasika-chivate) [inline docs/setting-up/server/ha.md:21]:
```suggestion
- PMM Server service is running (Docker, Podman, VM)
```

File: docs/setting-up/server/ha.md [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+- PMM on Kubernetes (see [Helm])
+- Pacemaker: a high-availability cluster resource manager
+
+## PMM on Kubernetes
+
+Kubernetes is an open-source system for automating containerized applications' deployment, scaling, and management.
+
+It helps PMM Administrator to avoid complexity in taking care of resources and shift this responsibility to the Kubernetes cluster:
+
+- networks: `Service` and `Ingress` provide DNS and IP access and connect PMM Server with clients and users
```
Comment (rasika-chivate) [inline docs/setting-up/server/ha.md:34]:
```suggestion
- Networks: `Service` and `Ingress` provide DNS and IP access and connect PMM Server with clients and users
```

File: docs/setting-up/server/ha.md [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+- storage: `PVC` ensures that storage is claimed only by one PMM Server and attached to it
```
Comment (rasika-chivate) [inline docs/setting-up/server/ha.md:35]:
```suggestion
- Storage: `PVC` ensures that storage is claimed only by one PMM Server and attached to it
```

File: docs/setting-up/server/ha.md [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+- PMM Server: `StatefulSet` runs PMM as a container and connects it to the correct storage and network
+
+PMM administrator could deploy on Kubernetes via [Helm] chart. And Kubernetes would make sure that PMM is available in case of different failures.
+
+Check our "How To": [Install PMM Server on Amazon EKS].
```
Comment (rasika-chivate) [inline docs/setting-up/server/ha.md:40]:
```suggestion
Check our **How To**: [Install PMM Server on Amazon EKS].
```

File: docs/setting-up/server/ha.md [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+## Pacemaker
+
+[Pacemaker] is a high-availability cluster resource manager.
+
+All of the currently supported distributions ship a high availability add-on/extension, which is based on the Pacemaker clustering stack, for example:
+
+- [Red Hat High Availability Add-On]
+- [SUSE Linux Enterprise High Availability Extension]
+
+Please check your distribution documentation on Pacemaker HA as well as [Pacemaker] documentation.
```
Comment (rasika-chivate) [inline docs/setting-up/server/ha.md:50]:
```suggestion
Check your distribution documentation on Pacemaker HA as well as [Pacemaker] documentation.
```

File: docs/setting-up/server/ha.md [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+
+Check our "How To": [Configure Pacemaker to manage PMM HA cluster](../../how-to/pacemaker.md).
```
Comment (rasika-chivate) [inline docs/setting-up/server/ha.md:52]:
```suggestion
Check our **How To**: [Configure Pacemaker to manage PMM HA cluster](../../how-to/pacemaker.md).
```

**Review by rasika-chivate — CHANGES_REQUESTED** [review]

File: docs/how-to/pacemaker.md [hunk 11]
```diff
@@ -0,0 +1,386 @@
...
+
+![](../_images/PMM-HA-Active-Standby.png "PMM Server, Two-Node (Active-Standby) High Availability Cluster")
+
+In case of failures, Pacemaker should make sure that IP address migrates to another node and shared storage would be mounted there as well for the passive PMM Server that would be activated in case of a failover.
+
+Fencing mechanism protects and restricts access to resources to avoid corruption and collisions between cluster nodes.
+
+In this example we would:
+
+- turn off the fencing for demonstration purposes and as it depends on HW that is used
```
[the remark this answers is not in the document]
Comment (denisok) [reply docs/how-to/pacemaker.md:33]: yes, should be Hardware

**Review by rasika-chivate — CHANGES_REQUESTED** [review]

File: docs/how-to/pmm-eks.md [hunk 12]
```diff
@@ -0,0 +1,273 @@
...
+    helm repo add percona https://percona.github.io/percona-helm-charts/
+    ```
+
+2. Update your local repository:
+
+    ```sh
+    helm repo update
+    ```
+
+3. Define [Parameters](https://github.com/percona/percona-helm-charts/tree/main/charts/pmm#parameters). An ELB load balancer. You can use an Network Load Balancer.
```
[the remark this answers is not in the document]
Comment (denisok) [reply docs/how-to/pmm-eks.md:227]: @anton-bystrov ^

File: docs/how-to/pacemaker.md [hunk 13]
```diff
@@ -0,0 +1,386 @@
...
+WantedBy=default.target
+
+EOF
+
+systemctl daemon-reload
+```
+
+Pre pull PMM Server image on each node for fast start of the service:
+```sh
+podman pull docker.io/percona/pmm-server:2.32.0
```
[the remark this answers is not in the document]
Comment (denisok) [reply docs/how-to/pacemaker.md:151]: yes

File: docs/how-to/pmm-eks.md (hunk 12 above)
[the remark this answers is not in the document]
Comment (anton-bystrov) [reply docs/how-to/pmm-eks.md:227]: We can use two kinds of balancers - ALB and NLB. Here want to describe how to install and use Network Load Balancer (NLB),

File: docs/how-to/pacemaker.md [hunk 14]
```diff
@@ -0,0 +1,390 @@
...
+## Add resources to the cluster
+
+With the following commands, we will add:
+
+* LVM HA that would be properly migrated and mounted by Pacemaker
+* VirtualIP that also would be migrated and added by the Pacemaker to the Active node
+* PMM Server systemd service that would start PMM Server with correct resources in place
+
+```sh
+pcs resource create pmm_lvm ocf:heartbeat:LVM-activate vgname=pmm_server vg_access_mode=system_id --group pmm
```
Comment (anton-bystrov) [inline docs/how-to/pacemaker.md:248]: May be typo here 'pmm_server'? Because in 'vgcreate --setautoactivation n pmm-server /dev/sda1' is 'pmm-server'.

Comment (denisok) [reply docs/how-to/pacemaker.md]: fixed! tnx!

**Review by anton-bystrov — APPROVED** [review]

**Comment by cezmunsta** [comment]
IMO `docs/how-to/pacemaker.md` should either be removed, or requires a highly conspicuous disclaimer that it is _not supported_.

**Comment by denisok** [comment]
It was decided to post pone this doc and don't write anything in a documentation about HA until it would be complete. @dygksy is going to create blog post about the HA phases instead.

**Pull request closed without merging**